src/main/javassist/util/proxy/ProxyFactory.java

```diff
@@ -1203,16 +1203,35 @@ private void getMethods(Map<String,Method> hash, Class<?> clazz, Set<Class<?>> v
             }
     }
 
-    private static boolean isOverloaded(int index, Method[] methods) {
+    private static boolean isDuplicated(int index, Method[] methods) {
         String name = methods[index].getName();
         for (int i = 0; i < methods.length; i++)
             if (i != index)
-                if (name.equals(methods[i].getName()))
+                if (name.equals(methods[i].getName()) && areParametersSame(methods[index], methods[i]))
                     return true;
 
         return false;
     }
-
+    
+    private static boolean areParametersSame(Method method, Method targetMethod) {
+        Class<?>[] methodTypes = method.getParameterTypes();
+        Class<?>[] targetMethodTypes = targetMethod.getParameterTypes();
+        int i = -1;
+        if (methodTypes.length == targetMethodTypes.length) {
+            for (Class<?> clz : methodTypes) {
+                i++;
```

Comment:
this is weird =/ why not just use a normal for statement with the `i` counter then?